modules/markup/csv/csv.go

```diff
@@ -77,29 +77,62 @@ func writeField(w io.Writer, element, class, field string) error {
 }
 
 // Render implements markup.Renderer
-func (Renderer) Render(ctx *markup.RenderContext, input io.Reader, output io.Writer) error {
+func (r Renderer) Render(ctx *markup.RenderContext, input io.Reader, output io.Writer) error {
 	tmpBlock := bufio.NewWriter(output)
+	maxSize := setting.UI.CSV.MaxFileSize
 
-	// FIXME: don't read all to memory
-	rawBytes, err := io.ReadAll(input)
+	if maxSize == 0 {
+		return r.tableRender(ctx, input, tmpBlock)
+	}
+
+	rawBytes, err := io.ReadAll(io.LimitReader(input, maxSize+1))
```

Comment:
If we can get file size from input, for example `input.(*os.File)`, there is no need to call io.ReadAll . It's really?

Reply:
If it is from git's "Blob", then it could know the size. However it would be much more complex, IMO current approach is also acceptable.

Reply:
So maybe we can give a size as a parameter and if it's zero, we can do this one?

Reply:
But it can be the next step.